pykern/pkcli/ci.py

```diff
@@ -73,6 +67,51 @@ def run():
     from pykern.pkcli import fmt, test
     from pykern import pkio
 
+    check_eof_newline()
     check_prints()
     fmt.diff(pkio.py_path())
     test.default_command()
+
+
+def _check_files(case):
+    def _check_eof_newline(file_path, res):
+        if pkio.read_text(file_path).split("\n")[-1] != "":
+            res.append(f"{file_path}")
+
+    def _check_prints(file_path, res):
+        for j, l in enumerate(pkio.read_text(file_path).split("\n"), start=1):
+            if re.search(_PRINT, l) and not re.search(_PRINT_OK, l):
+                res.append(f"{file_path}:{j} {l}")
+
+    def _error(m):
+        pkcli.command_error("{}: {}", case, m)
+
+    if case == "check_eof_newline":
+        d = PKDict(
```

Comment:
This PKDict could be a constant instead of having to map the names, and it could be a pkdict of pkdicts:

```py
PKDict(check_prints=PKDict(exclude_files=re.compile(
    r"/ext/|node_modules/|^run/|tests/|^venv/"
), 
etc.
```

Reply:
f2008e1c66a39ead6e0c01f83feddc1c1b0d258d